ts/annotations/types/BasicAnnotation.ts

```diff
@@ -210,26 +210,41 @@ class BasicAnnotation extends Annotation {
     public addControlPoints(): void {
         const options = this.options,
             controlPoints = BasicAnnotation.basicControlPoints,
-            langKey = options.langKey,
+            annotationType = this.basicType,
             optionsGroup = options.labels || options.shapes;
 
         optionsGroup.forEach(function (
             group: Highcharts.AnnotationControllableOptionsObject
         ): void {
-            if (langKey) {
-                // @todo langKey === 'label' / 'circle' / 'rectangle' ???
-                group.controlPoints = (controlPoints as any)[langKey];
-            }
+            group.controlPoints = (controlPoints as any)[annotationType];
         });
     }
 
+    public init(): void {
+        const options = this.options;
+
+        if (options.shapes) {
+            delete options.labelOptions;
+            if (options.shapes[0].type === 'circle') {
+                this.basicType = 'circle';
+            } else if (options.shapes[0].type === 'path') {
+                this.basicType = 'rectangle';
+            }
```

Comment:
For a sanity check, I would just use:
```js
            if (options.shapes[0].type === 'circle') {
                this.basicType = 'circle';
            } else {
                this.basicType = 'rectangle';
            }
```

To avoid situation when `basicType` will be undefined.